rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,175 @@
+    \item Variable \texttt{input\_metric} is of type \texttt{AbsoluteDistance<f64>}.
+    \item Variable \texttt{epsilon} is of type \texttt{float}
+    \item Variable \texttt{delta} is of type \texttt{float}
+\end{itemize}
+
+\subsection*{Pseudocode}
+
+\lstinputlisting[language=Python,firstline=2]{./pseudocode/make_tulap.py}
+
+\subsection*{Postcondition}
+
+\validMeasurement{\texttt{(input\_domain, input\_metric, epsilon, delta)}}{\\ \texttt{make\_tulap}}
+
+\section{Proof}
+
+\begin{proof} 
+\hfill
+\begin{enumerate}
+    \item \textbf{(Privacy guarantee.)} 
+
+\begin{tcolorbox}
+    \begin{note}[Proof relies on correctness of the Tulap sampler]
+    The following proof makes use of the following lemmas that asserts the correctness of the Tulap sampler function.
+    \begin{lemma}
+        If system entropy is insufficient, \texttt{pinpoint} raises an error. 
+        Otherwise, \texttt{pinpoint}, the function that narrows the Tulap PSRN, 
+        returns sample from the Tulap distribution.
+    \end{lemma}
+    \end{note}
+\end{tcolorbox}
+
+\texttt{pinpoint} can only fail when the pseudorandom byte generator used in its implementation fails due to lack of system entropy. 
+This is usually related to the computer's physical environment and not the dataset. 
+The rest of this proof is conditioned on the assumption that \texttt{pinpoint} does not raise an exception. \\
+
+A canonical noise distribution (CND)\parencite{awan2023canonical}, 
+captures whether a real-valued distribution is perfectly tailored to satisfy $f$-DP\parencite{dong2019gaussian}. 
+We formalize this in Definition \ref{def2}\parencite{awan2023canonical}. 
+\begin{defn}\label{def2}  % Def 3.1
+Let $f$ be a symmetric nontrivial tradeoff function. A {continuous} distribution function $F$ is a \emph{canonical noise distribution} (CND) for $f$ if 
+\begin{enumerate}[(1)]
+    % Mike: there was an \mscr after "S:". What is it for?
+    \item for every statistic $S: X^n\rightarrow \mathbb{R}$ with sensitivity $\Delta>0$, and $N\sim F(\cdot)$, 
+        the mechanism $S(X) + \Delta N$ satisfies $f$-DP. Equivalently, for every $m\in [0,1]$, $T(F(\cdot),F(\cdot-m))\geq f$,
+    \item $f(\alpha)=T(F(\cdot),F(\cdot-1))(\alpha)$ for all $\alpha \in (0,1)$,
+    \item $T(F(\cdot),F(\cdot-1))(\alpha) = F(F^{-1}(1-\alpha)-1)$ for all $\alpha \in (0,1)$,
+    \item $F(x) = 1-F(-x)$ for all $x\in \mathbb{R}$; that is, $F$ is the cdf of a random variable which is symmetric about zero.
+\end{enumerate}
+\end{defn}
+
+\begin{defn} \label{def1}  % Def 3.7
+Let $f$ be a symmetric nontrivial tradeoff function, and let {$c\in [0,1/2)$} be the unique fixed point of $f$: $f(c)=c$. 
+We define $F_f:\mathbb{R}\rightarrow \mathbb{R}$ as  \[ F_f(x) = \begin{cases}
+    f(1-F_f(x+1))&x<-1/2\\
+    c(1/2-x) + (1-c)(x+1/2)&-1/2\leq x\leq 1/2\\
+    1-f(F_f(x-1))&x>1/2.\\
+\end{cases}\]
+\end{defn}
+In Definition \ref{def1} \parencite{awan2023canonical}, 
```

Comment:
"Definition~\ref{...}"

Do this for the other definition/theorems/lemmata/propositions/claims, as well.